### PR TITLE
Feat: Android ad config

### PR DESCRIPTION
**Pull request by aronshamash**
- Populate ad config for Android.
- Corresponding Android Smartphone PR: https://github.com/newsuk/nu-projectd-times-smartphone-android/pull/420

### CODE REVIEW

File: android-app/pages/article.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import PropTypes from "prop-types";
 import { NativeModules } from "react-native";
 import { Article } from "@times-components/pages";
+import DeviceInfo from "react-native-device-info";
```
Comment (tuncaulubilge) [inline android-app/pages/article.js]: We should be getting rid of this dependency eventually. I'll check if I can pass the device id from the native app.

Comment (tuncaulubilge) [reply android-app/pages/article.js]: Created an issue to remove it completely: https://github.com/newsuk/times-components/issues/1126

Comment (aronshamash) [reply android-app/pages/article.js]: We can get from native app.

File: android-app/pages/article.js [hunk 2]
```diff
@@ -15,41 +16,69 @@ const {
...
+  cookieIamTgt: config.cookieIamTgt,
+  isLoggedIn: config.isLoggedIn,
   platform: "mobile"
-};
+});
+
+const ArticleView = ({
+  articleId,
+  sectionName,
+  appVersion,
```
Comment (tuncaulubilge) [inline android-app/pages/article.js]: We should get this, and all the props below from `config`. I'm moving these inside config in the native app

File: android-app/pages/article.js [hunk 3]
```diff
@@ -15,41 +16,69 @@ const {
...
-  deviceId: "",
-  deviceIdHash: "",
-  environment: "",
-  isLoggedIn: true,
+  sectionName: config.sectionName,
+  appVersion: config.appVersion,
+  operatingSystem: "Android",
+  operatingSystemVersion: config.operatingSystemVersion,
+  environment: config.environment,
+  deviceId: DeviceInfo.getUniqueID(),
```
Comment (tuncaulubilge) [inline android-app/pages/article.js]: Should be available through `config.deviceId` soon

File: android-app/pages/article.js [hunk 4]
```diff
@@ -15,41 +15,48 @@ const {
 } = NativeModules.ArticleEvents;
 const ArticlePageView = Article(config)(fetch);
 
-const platformAdConfig = {
+const getPlatformAdConfig = config => ({
```
Comment (tuncaulubilge) [inline android-app/pages/article.js]: This needs to be renamed, as it clashes with the upper level config (` const config = NativeModules.ReactConfig;`)

File: android-app/pages/article.js [hunk 5]
```diff
@@ -15,41 +15,41 @@ const {
...
-  cookieAcsTnl: "",
-  cookieIamTgt: "",
-  deviceId: "",
-  deviceIdHash: "",
-  environment: "",
-  isLoggedIn: true,
-  platform: "mobile"
-};
+const ArticleView = ({ articleId, sectionName }) => {
+  const adConfig = {
```
Comment (craigbilner) [inline android-app/pages/article.js]: is there value in binding this on every function call?

File: packages/pages/src/client/ad-targeting-config.js [hunk 6]
```diff
@@ -1,15 +1,13 @@
 export default (platformAdConfig, articleAdConfig) => ({
-  networkId: platformAdConfig.adUnit,
-  adUnit: platformAdConfig.networkId,
+  networkId: platformAdConfig.networkId,
```
Comment (craigbilner) [inline packages/pages/src/client/ad-targeting-config.js]: shouldn't this have  broken a test?

**Pull request merged**